### PR TITLE
Update timeout documentation

### PR DESCRIPTION
**Pull request by eiri**
This PR adds a more accurate description for configured timeout in python and adds details on timeout defaults across the languages.

Closes:
- https://github.ibm.com/arf/planning-sdk-squad/issues/3164
- https://github.ibm.com/arf/planning-sdk-squad/issues/2836

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1770,6 +1770,8 @@ When constructing a Context, you receive two values - the context itself, along
 The returned cancel function can also be invoked in order to cancel an in-flight request.
 Note also that by using the `context.Context` instance with the `<operation-name>WithContext` method,
 the timeout applies only to a single operation invocation.
+
+By default the client has no request timeout, but a 30s connect timeout is set by default.
```
Comment (dpopp07) [inline README.md:1774]: I think this working might be a little confusing. Are you trying to document that the underlying client package we use doesn't have a default but that the SDK core is setting a default? Is it important to clarify what the underlying client package's default behavior is? Perhaps I'm missing (or forgetting) some context as to the value of thatclarification

File: README.md [hunk 2]
```diff
@@ -1786,6 +1788,8 @@ myService.setClient(client);
 ```
 
 The request timeout of 10 seconds will be used in each operation invoked using `myService1`.
+
+By default the client has no request timeout, but a 60s connect, a 90s read and a 60s write timeout are set by default.
```
Comment (padamstx) [inline README.md:1792]:
```suggestion
By default, the client has a 60-second connect timeout, a 90-second read timeout and a 60-second write timeout.  The default client does not set a `call timeout` (which covers the entire HTTP call... including the connection, write and read steps), but you can set that as in the example above.
```

File: README.md [hunk 3]
```diff
@@ -1802,17 +1806,21 @@ const myService = new ExampleServiceV1({
...
+
+By default the client has no request timeout.
 </details>
 <details><summary>Python</summary>
 As mentioned in the "Configuring the HTTP Client" section above, you can configure the options
 in the http client as in this example:  
 
 ```python
-# Configure a 10-second request timeout.
+# Configure a 10-second read timeout.
```
[the remark this answers is not in the document]
Comment (eiri) [reply README.md:1817]: Yes, I know python's trying to be smart here and I've mulled over how to express it better for a while. In your wording it sounds like 10 seconds applied to both operations, so the total timeout will be 20 secs, while it's actually be half of that.

File: README.md [hunk 4]
```diff
@@ -1802,17 +1806,21 @@ const myService = new ExampleServiceV1({
...
 my_service.set_http_config({'timeout': 10})
 ```
 
-The request timeout of 10 seconds will be used in each operation invoked using `my_service`.
+The read timeout of 10 seconds, which includes the connect timeout, will be used in each operation invoked using `my_service`.
```
[the remark this answers is not in the document]
Comment (eiri) [reply README.md:1821]: I don't think this is correct. `Requests` uses `urllib3` under the hood and quoting its docs
> (total) - This combines the connect and read timeouts into one; the read timeout will be set to the time leftover from the connect attempt. 

This described in the referenced issues and in [this ref ticket](https://github.ibm.com/arf/planning-sdk-squad/issues/2852)

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (padamstx) [reply README.md:1817]: My reading of [this page](https://requests.readthedocs.io/en/latest/user/advanced/#timeouts) is such that if you specify `timeout` as a single value, that value applies separately to connections and read operations.   So setting it to `10` means that a connection must be established before 10 secs elapses.  Then a separate/independent 10-second timeout is associated with the read operation.   I'm interpreting the docs this way due to the fact that a tuple (e.g. `timeout=(10, 20)`) can be used to specify the separate "connect" and "read" timeout values.  So it follows logically that `timeout=10` is equivalent to `timeout=(10, 10)` (i.e. connect timeout of 10s, followed by a read timeout of 10s).

File: README.md [hunk 5]
```diff
@@ -1802,17 +1806,21 @@ const myService = new ExampleServiceV1({
...
+
+By default the client has no request timeout, but 60s read timeout, that includes the connect timeout is set by default.
```
[the remark this answers is not in the document]
Comment (eiri) [reply README.md:1823]: Again, 60 secs is not as much applied to both connection and read operation as shared between them. Also they do not equal the request time, hence the original wording. Here is another quote from `urllib3`
> the read and total timeouts only measure the time between read operations on the socket connecting the client and the server, not the total amount of time for the request to return a complete response.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (padamstx) [reply README.md:1817]: Upon further review :)...  I re-read the "requests" documentation, plus I looked through some of the requests source code, along with some of the urllib3 documentation and source code, and it appears that the requests library simply passes through the timeout value to the urllib3 library.   I'll update my comments above...

File: README.md (hunk 4 above)
Comment (padamstx) [inline README.md:1821]:
```suggestion
This timeout setting will be used in each operation invoked using `my_service`.
```

**Review by padamstx — CHANGES_REQUESTED** [review]
I've removed some of my prior comments after more digging in "requests" and "urllib3".

**Review by padamstx — APPROVED** [review]
I've re-reviewed and everything LGTM.

**Pull request merged**